modules/dnn/test/test_caffe_importer.cpp

```diff
@@ -55,6 +55,43 @@ static std::string _tf(TString filename)
     return (getOpenCVExtraDir() + "/dnn/") + filename;
 }
 
+static bool readFileInMemory(const string& filename, string& content)
+{
+    std::ios::openmode mode = std::ios::in | std::ios::binary;
+    std::ifstream ifs(filename.c_str(), mode);
+    if (!ifs.is_open())
+        return false;
+
+    content.clear();
+
+    ifs.seekg(0, std::ios::end);
+    content.reserve(ifs.tellg());
+    ifs.seekg(0, std::ios::beg);
+
+    content.assign((std::istreambuf_iterator<char>(ifs)),
+                    std::istreambuf_iterator<char>());
+
+    return true;
+}
+
+TEST(Test_Caffe, memory_read)
```

Comment:
I think it'd be better to test an accuracy of one of the models too.

Reply:
in the `memory_read` test case (as I have done for `Test_TensorFlow`) or in a separate test case ?

Reply:
In example, replace `TEST(Reproducibility_AlexNet, Accuracy)` to
```cpp
typedef TestWithParam<tuple<bool> > Reproducibility_AlexNet;
TEST_P(Reproducibility_AlexNet, Accuracy) {
  float readFromMemory = get<0>(GetParam());
  ...
}

INSTANTIATE_TEST_CASE_P(Test_Caffe, Reproducibility_AlexNet, Values(
  Bool()
));
```

Reply:
done